name-service/program/tests/functional.rs

```diff
@@ -175,6 +176,58 @@ async fn test_name_service() {
     .unwrap();
     println!("Name Record Header: {:?}", name_record_header);
 
+    let data = "hello".as_bytes().to_vec();
+    let update_instruction = update(
+        program_id,
+        space as u32,
+        data,
+        name_account_key,
+        sol_subdomains_class.pubkey(),
+        Some(name_record_header.parent_name),
+    )
+    .unwrap();
+
+    sign_send_instruction(
+        &mut ctx,
+        update_instruction.clone(),
+        vec![&sol_subdomains_class],
+    )
+    .await
+    .unwrap_err();
+
+    #[allow(clippy::integer_arithmetic)]
+    let new_space = space * 2;
+    let payer_key = ctx.payer.pubkey();
+    let realloc_instruction = |space| {
+        realloc(
+            program_id,
+            payer_key,
+            name_account_key,
+            payer_key,
+            space as u32,
+        )
+        .unwrap()
+    };
+
+    sign_send_instruction(&mut ctx, realloc_instruction(new_space), vec![])
+        .await
+        .unwrap();
+
+    // warping slot to resend update ix without dropping duplicate txn
+    #[allow(clippy::integer_arithmetic)]
+    let new_slot = ctx.banks_client.get_root_slot().await.unwrap() + 32;
+    ctx.warp_to_slot(new_slot).expect("warp failed");
```

Comment:
Rather than warping, you can simply fetch a new blockhash using `ctx.banks_client.get_new_latest_blockhash(&ctx.last_blockhash)` and then setting that onto the context. Or make a different instruction

Reply:
I think the reason I've been against doing that in the past is because last I checked its actually just waiting for a new block + blockhash to happen. On a large scale this can slow stuff down quite a bit. For this one instance though you're totally right. I'll change that.